src/ranking.jl

```diff
@@ -171,3 +171,170 @@ Missing values are assigned rank `missing`.
+
+# use `Ref` to treat vector `v3` as a scalar during broadcasting.
+```julia
+julia> quantilerank.(Ref(v3), [4, 8])
+2-element Vector{Float64}:
+ 0.3333333333333333
+ 0.8888888888888888
+```
+"""
+function quantilerank(v, value; method::Symbol=:inc)
+    # checks
+    value isa Number && isnan(value) &&
+        throw(ArgumentError("value cannot be NaN"))
+    any(x -> ismissing(x) || (x isa Number && isnan(x)), v) &&
+        throw(ArgumentError("`v` cannot contain missing or NaN entries"))
+
+    count_less = count_equal   = n = 0
+    last_less  = first_greater = value
+    for x in v
+        if x == value
+            count_equal += 1
+        elseif x < value
+            count_less += 1
+            if last_less == value || last_less < x
+                last_less = x
+            end
+        else
+            if first_greater == value || first_greater > x
+                first_greater = x
+            end
+        end
+        n += 1
+    end
+
+    n == 0 && throw(ArgumentError("`v` is empty. Insert a non-empty vector"))
+    n == 1 && throw(ArgumentError("`v` has only 1 value. Use a vector with more elements"))
+
+    if method == :inc
+        if last_less == value
+            return 0.0
+        elseif count_equal > 0
+            return count_less / (n - 1)
+        elseif first_greater == value
+            return 1.0
+        else
+            lower = (count_less - 1) / (n - 1)
+            upper = (count_less) / (n - 1)
+            ratio = (value - last_less) / (first_greater - last_less)
+            return lower + ratio * (upper - lower)
+        end
+    elseif method == :exc
+        if count_less == 0 && count_equal == 0
+            return 0.0
+        elseif count_less == 0
+            return 1.0 / (n + 1)
+        elseif count_equal > 0
+            return (count_less + 1) / (n + 1)
+        elseif first_greater == value
+            return 1.0
+        else
+            lower = (count_less) / (n + 1)
+            upper = (count_less + 1) / (n + 1)
+            ratio = (value - last_less) / (first_greater - last_less)
+            return lower + ratio * (upper - lower)
+        end
+    elseif method == :compete
+        if value > maximum(v)
+            return 1.0
+        elseif value ≤ minimum(v) 
+            return 0.0
+        else
+            value ∈ v && (count_less += 1)
+            return (count_less - 1) / (n - 1)
+        end 
+    elseif method == :tied
+        return (count_less + count_equal/2) / n
+    elseif method == :strict
+        return count_less / n
+    elseif method == :weak
+        return (count_less + count_equal) / n
+    else
+        throw(ArgumentError("method=:$method is not valid. Use :inc, :exc, :compete, :tied, :strict or :weak."))
+    end
+end
+
+"""
+    percentrank(v, value; method=:inc)
+
+Return the `q`th percentile of a collection `value`, i.e. [`quantilerank`](@ref) * 100.
+"""
+percentrank(v, value; method::Symbol=:inc) = quantilerank(v, value, method=method) * 100
```

Comment:
```suggestion
percentrank(v, value; method::Symbol=:inc) = quantilerank(v, value, method=method) * 100

```

Reply:
In conversation above, @nalimilan told me that in StatsBase.jl they don't put this line at end.

Reply:
There's probably a misunderstanding. :-) I don't really have an opinion about ending line breaks. Maybe I just suggested to avoid changing the current state if that's not necessary.

Reply:
looking through other files in the package, I saw that it has an blank line at the end of the file.

So I will add this end line break to keep the pattern.